docs/distributions/gentoo/installation.md

```diff
@@ -0,0 +1,63 @@
+5. Follow the [Gentoo Handbook](https://wiki.gentoo.org/wiki/Handbook:AMD64/Installation/Disks)
+    1. Instead of making a new EFI partition for Gentoo, you should instead mount `/dev/nvme0n1p1` to `/mnt/gentoo/boot/efi`
+    2. Make sure that you have deleted the partition you made ahead of time, or else you'll lose important data
+6. Continue through the Handbook until chapter "Configuring the kernel"
+    1. In order to have the Linux Kernel support your device the most, you'll have to manually apply patches to the Linux Kernel.
+    2. Run these commands in the Terminal:
+
+    ```bash
+    emerge sys-kernel/gentoo-sources
+    eselect kernel list # there should be only one option that shows up
+    eselect kernel set 1
+    ls -l /usr/src/linux # to check if the symlink succeded
+    ```
+
+    1. Git clone [these patches](https://github.com/Redecorating/mbp-16.1-linux-wifi) to `/linux-patches`
+    2. If needed, git checkout to an older tag with the current gentoo-sources kernel version
+    3. Run these commands to grab apple-bce and apple-ibridge:
+
+    ```bash
+    git clone https://github.com/t2linux/apple-bce-drv /usr/src/apple-bce
+    git clone https://github.com/t2linux/apple-ib-drv /usr/src/apple-ibridge
+    for i in apple-bce apple-ibridge; do
+    mkdir /usr/src/linux/drivers/staging/$i 
+    cp -r /usr/src/$i/* /usr/src/linux/drivers/staging/$i/
+    ```
+
+    1. Apply all patches from `/linux-patches` to `/usr/src/linux`.
+    2. It's recommended to use the config from the patches repo instead of the default config from `gentoo-sources`. If you do this, please make sure to set any filesystem drivers you want to use (like for ext4) to be built-in instead of being a module
+    3. Build with `make all`. If you want to speed up the build process, you might want to set `-j` to something like 2.
+7. Before finishing, if you want to connect to the internet later, now is a good time to install `NetworkManager` and optionally `iwd`.
+8. Continue through the Handbook until chapter "Configuring the bootloader"
+    1. Install grub by using `emerge --ask --verbose sys-boot/grub`
+    2. Edit the config file `/etc/default/grub` with your favorite text editor of choice (i.e. vim or nano)
+    3. On the line with `GRUB_CMDLINE_LINUX`, append the following parameters: `intel_iommu=on iommu=pt pcie_ports=compat`
+    4. Run `grub-install --target=x86_64-efi --efi-directory=/boot/efi --bootloader-id=GRUB --no-nvram --removable` to insall Grub.
+    5. Run `grub-mkconfig -o /boot/grub/grub.cfg` to make the config files for Grub
+    6. Make nvram/efivars automatically remount as readonly, since writing to them causes a kernel panic: `echo efivarfs /sys/firmware/efi/efivars efivarfs ro,remount,nofail 0 0 >> /etc/fstab`. If this for some reason does not work, then add `efi=noruntime` to the kernel paramters in `/etc/default/grub`
```

Comment:
It'd probably be simpler to manually also take the 0101 patch from the main branch of my repo and apply after the other patches, it should apply to 5.15. (but this would no longer be necessary once gentoo updates to 5.16)

Reply:
But the thing is, the 0101 patch would already be applied. Are you saying that I should tell people building newer kernels to remove that patch?

Reply:
I added the 0101 patch in 5.16-2, so it wouldn't be there if you checkout v5.15.13-1. When gentoo updates to newer kernels, people would checkout a commit that already has the 0101 patch, so they would not need to manually get it (it would be applied with the other patches)

Reply:
Oh I see